modules/xp/weekly.ts

```diff
@@ -140,6 +141,30 @@ export default async function getWeekly(nextWeeklyDate: Date) {
 		]);
 	}
 
+	const customRoles = (await config.guild.roles.fetch()).filter((role) =>
+		role.name.toLowerCase().includes("✨ "),
+	);
+	weeklyWinners.forEach(async (weeklyWinner) => {
+		await config.guild.members.fetch(weeklyWinner.user).then(async (guildMember) => {
+			const filtered = customRoles.filter((customRole) => {
```

Comment:
move L123-125 of roles.ts to a function (search for PREFIX and replace other usages as well while you're at it)

in here, check if they qualify
if not, run that function and just delete the role

Reply:
maybe resolve it?

Reply:
![](https://images.app.goo.gl/rcjEoJR2BHKZZmoH7)

> search for PREFIX and replace other usages as well while you're at it

Also, please revert the package.json changes. You will need to disable actions so they stop being recreated. In the future, don't use the main branch and you won't have a problem

After that, I can merge this

Reply:
Ah only if I was on my lappy